### ISSUE TITLE
Add `take` and / or indexing operations

### ISSUE THREAD

**Issue by shcheklein** [iterative/datachain#374]
We need `take(n)` or `take([1,2,3])` (or just `[]`) operation to avoid constructs like this in our examples:

```python
mysterious_dogs = list(no_dogs_detected.collect("file"))
mysterious_dogs[11].read()
```

we can't fetch into memory the whole dataset to read a single file

- Indexing might be complicated (do we need to support column names there and all that stuff that pandas can do)?
- Do we have an order in datachain, is it stable?

**Comment by dmpetrov** [comment]
.limit(N).collect() should be enough based on my understanding of the problem. I'd not introduce a special command now.

Order is not guaranteed (which is usual for DW) until ordering commands is specified. Technically it can be done using ID.

**Comment by shcheklein** [comment]
if you need to get a single 1000th element you would have to bring all 1000 elements into memory, right?

**Comment by shcheklein** [comment]
> Order is not guaranteed (which is usual for DW) until ordering commands is specified. Technically it can be done using ID.

right, it means probably that in that specific example semantic is undefined:

```
mysterious_dogs = list(no_dogs_detected.collect("file"))
mysterious_dogs[11].read()
```

no guarantee that 11 returns the same element every time?

probably, should be rewritten to pick an element by its name / path to be safe

**Comment by dmpetrov** [comment]
That's right.

Ideally, order() without params should order by the hidden ID.

For range, 1000-1010 we can introduce offset in order() which is standard in sql: dc.order(offset=1000).limit(10).collect().

This is how it suppose to work. Take() is a syntax sugar and can be introduce later if there enough demand.

**Comment by shcheklein** [comment]
> For range, 1000-1010 we can introduce offset in order() which is standard in sql: dc.order(offset=1000).limit(10).collect().

yep, that's totally fine if it allows to get a specific / small subset of elements

**Comment by shcheklein** [comment]
I think we already have `sample`, `offset`, `limit` - it should be enough for now.
